### PR TITLE
Add script to use app registration instead of user access

### PR DESCRIPTION
**Pull request by SrinidhiRaghavendra-Rubrik**
The current script uses admin user access to login to exchange powershell to run commands.
Adding a new access method to login using application credentials with this PR.

Tested on a sandbox subscription.
1. Incorrect option selection:
Screenshot 2024-12-09 at 1 03 34 PM

2. User access (login through the user account on browser):
Screenshot 2024-12-09 at 1 05 07 PM

3. App access:
Screenshot 2024-12-09 at 1 08 01 PM

### CODE REVIEW

File: Get-RubrikM365SizingInfo.ps1 [hunk 1]
```diff
@@ -283,6 +283,31 @@ function Get-RecoverableItemsInfo {
     return $RIFItemsStatistics
 }
 
+function Get-AccessToken {
+    param (
+        [string]$ClientId,
+        [string]$ClientSecret,
+        [string]$TenantId,
+        [string]$Scope = "https://outlook.office365.com/.default"
```
Comment (iirohit) [inline Get-RubrikM365SizingInfo.ps1:291]: just curious, will we use the app credentials only for the mailbox sizing as scope is specific to outlook?

Also, is there a way we are letting customer know about the permission to add in this app?

File: Get-RubrikM365SizingInfo.ps1 [hunk 2]
```diff
@@ -315,26 +340,64 @@ if ($AzureAdRequired) {
...
-      $errorException = $_.Exception
-      $errorMessage = $errorException.Message
-      Write-Host "[ERROR] Unable to Connect to the Microsoft Exchange PowerShell Module: $errorMessage"
+        Connect-MgGraph -TenantId $tenantId -ClientSecretCredential $clientSecretCredential
     }
-  }
+    catch {
+        $errorException = $_.Exception
+        $errorMessage = $errorException.Message
+        Write-Host "[ERROR] Unable to Connect to the Microsoft Graph PowerShell Module: $errorMessage"
```
Comment (iirohit) [inline Get-RubrikM365SizingInfo.ps1:360]: Do we need to terminate it here?

File: Get-RubrikM365SizingInfo.ps1 (hunk 1 above)
Comment (SrinidhiRaghavendra) [reply Get-RubrikM365SizingInfo.ps1:291]: This token is for connecting to ExchangeOnline only for archive mailbox and recoverable items data. For other snappables, we have the app permission 'Reports.Read.All' to get the required data.
For that, we are using `Connect-MgGraph` in line 355 to get the required data from the reports API.

Comment (SrinidhiRaghavendra) [reply Get-RubrikM365SizingInfo.ps1:291]: Have added that required permissions in README.md.

**Review by Draper1 — APPROVED** [review]

**Review by iirohit — APPROVED** [review]
Looks good to me. Can you please also add test plan? Have we tested with all the different options? Would be good to add screenshot as well.

**Pull request merged**